app/models/ip_block.rb

```diff
@@ -23,7 +23,7 @@ class IpBlock < ApplicationRecord
     sign_up_requires_approval: 5000,
     sign_up_block: 5500,
     no_access: 9999,
-  }
+  }, validate: true
```

Comment:
>  Do we need a migration which converts any values not in the valid options into one of them?

I don't think so. Since we're using a hash with integer values, Rails throws an `ArgumentError` when we attempt to instantiate an `IpBlock` object with an invalid severity. It does not allow any record to be persisted in the database with an invalid value.

```rb
irb(main):004> IpBlock.create!(ip: '151.0.32.55', severity: :invalid)
/home/daniel/.rbenv/versions/3.2.3/lib/ruby/gems/3.2.0/gems/activerecord-7.1.3/lib/active_record/enum.rb:206:in `assert_valid_value': 'invalid' is not a valid severity (ArgumentError)

          raise ArgumentError, "'#{value}' is not a valid #{name}"
````

I don't think there should be any invalid value in the database. Unless I'm missing something here 😅.

Reply:
Yeah - I noted similar opinion on the other issue. I think it's wildly unlikely for that exact reason (prior arg error) and it would not be irresponsible to just ignore it and not bother ... but it's possible (direct sql updates?)

Reply:
> but it's possible (direct sql updates?)

Yeah, that's a possibility. I tested it locally to check its behavior in that scenario. If an `IpBlock` has an invalid severity value in the database, such as -1000, its severity is considered `nil`. This contradicts the model validation that checks for the presence of severity. I'm not sure if it could potentially break anything in the codebase, though.

Reply:
I opted for a migration to keep the values in range (https://github.com/hendrixfan/mastodon/blob/add-domain-blocks-enum-validation/db/migrate/20240220082519_fix_invalid_domain_block_severities.rb). Maybe it would help here too.